src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -107,27 +110,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     // creates arraylist and starts query
-    ArrayList<String> groceryList = new ArrayList<String>();
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     long familyID = (long) userInfoEntity.getProperty(FAMILY_ID);
     Query groceryQuery = new Query(GROCERY)
       .setFilter(new Query.FilterPredicate(FAMILY_ID, Query.FilterOperator.EQUAL, familyID));   
     PreparedQuery familyGrocery= datastore.prepare(groceryQuery);
-    groceryList = checkGroceries(familyGrocery);   
+    groceryList = checkGroceries(familyGrocery, userEmail);   
```

Comment:
since the instantiated value in 113 is never used, just do

`ArrayList<Grocery> = checkGroceries(...)` here

Also, I would suggest renaming checkGroceries to `fetchGroceries` and moving the PreparedQuery inside it (so that the function encapsulates all the datastore logic as well)

Reply:
That value is used

Reply:
Sorry for the lack of clarity; I also realized I typoed in my comment (late Friday afternoon comments!).

The variable `groceryList` is used, but the value assigned on 113 (`new ArrayList<Grocery>`) is never used. The value is directly overridden on 118 by the return value of `checkGroceries`.

My suggestion is to thus to remove line 113.

Then, on line 118, you can just do `ArrayList<Grocery> groceryList = checkGroceries(...)`.

In general, variable declarations should be close to where they're actually assigned a useful value (and we should avoid instantiating empty arraylists that we never use).